sdk/digitaltwins/Azure.DigitalTwins.Core/tests/AdtQueryBuilderTests/WhereLogicTests.cs

```diff
@@ -176,14 +176,78 @@ public void WhereLogic_IsOfType_Object()
         }
 
         [Test]
-        public void WhereLogic_MultipleQueryies()
+        public void WhereLogic_MultipleQueryies_And()
         {
             var query = new WhereLogic(null);
             query.Compare("Temperature", QueryComparisonOperator.Equal, 50)
+                .And()
                 .IsDefined("Humidity");
             query.GetQueryText()
                 .Should()
                 .Be("Temperature = 50 AND IS_DEFINED(Humidity)");
         }
+
+        [Test]
+        public void WhereLogic_MultipleQueryies_Or()
+        {
+            var query = new WhereLogic(null);
+            query.Compare("Temperature", QueryComparisonOperator.Equal, 50)
+                .Or()
+                .IsDefined("Humidity");
+            query.GetQueryText()
+                .Should()
+                .Be("Temperature = 50 OR IS_DEFINED(Humidity)");
+        }
+
+        [Test]
+        public void WhereLogic_NestedQueries()
+        {
+            var query = new WhereLogic(null);
+            query.Compare("Temperature", QueryComparisonOperator.Equal, 50)
+                .And()
+                .IsTrue(q => q
+                   .IsOfType("Humidity", AdtDataType.AdtNumber)
+                   .And()
+                   .IsOfType("Humidity", AdtDataType.AdtPrimative));
+            query.GetQueryText()
+                .Should()
+                .Be("Temperature = 50 AND (IS_NUMBER(Humidity) AND IS_PRIMATIVE(Humidity))");
+        }
+
+        [Test]
+        public void WhereLogic_NestedWithinNestedQueries()
+        {
+            var query = new WhereLogic(null);
+            query.Compare("Temperature", QueryComparisonOperator.Equal, 50)
+                .And()
+                .IsTrue(q => q
+                   .IsOfType("Humidity", AdtDataType.AdtNumber)
+                   .And()
+                   .IsTrue(q => q
+                        .IsOfModel("dtmi:example:room;1", true)
+                        .Or()
+                        .IsOfType("isOccupied", AdtDataType.AdtBool)));
+            query.GetQueryText()
+                .Should()
+                .Be("Temperature = 50 AND (IS_NUMBER(Humidity) AND (IS_OF_MODEL('dtmi:example:room;1', exact) OR IS_BOOL(isOccupied)))");
```

Comment:
if you can somehow declare the expected result in a format that makes it clear how the operators are separated with parenthesis, it would be awesome.

Reply:
```
Temperature = 50 
AND 
(
    IS_NUMBER(Humidity) 
    AND 
    (
        IS_OF_MODEL('dtmi:example:room;1', exact) 
        OR 
        IS_BOOL(isOccupied)
    )
)
```